### PR TITLE
Do not swallow exceptions thrown while parsing settings

### LINKED ISSUE #13028: straight up garbage in elasticsearch.yml does not cause error or even warning.
I added this:
```
...
# action.destructive_requires_name: true

SKDFLK@$#L%@KL#%L#@$#@L$ #L$@$ #L@K$#L $L $K#L#@L $#L
!!@!@$(#%#)(@)% #(%)
#(%#@)%@#)% (@#%()
()#%@#% (@ )%@%(@#)% @( %)@ %(@)
)(%)@()(%)()(#%)@#
rmuir@beast:~/workspace/
```

Nothing happens.

### PR DESCRIPTION
**Pull request by jasontedor**
This commit fixes an issue that was causing Elasticsearch to silently
ignore settings files that contain garbage. The underlying issue was
swallowing an `SettingsException` under the assumption that the only
reason that an exception could be throw was due to the settings file
not existing (in this case the `IOException` would be the cause of the
swallowed `SettingsException`). This assumption is mistaken as an
`IOException` could also be thrown due to an access error or a read
error. Additionally, a `SettingsException` could be thrown exactly
because garbage was found in the settings file. We should instead
explicitly check that the settings file exists, and bomb on an
exception thrown for any reason.

Closes #13028

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java [hunk 1]
```diff
@@ -114,10 +114,9 @@
...
                 for (String allowedSuffix : ALLOWED_SUFFIXES) {
-                    try {
-                        settingsBuilder.loadFromPath(environment.configFile().resolve("elasticsearch" + allowedSuffix));
-                    } catch (SettingsException e) {
-                        // ignore
+                    Path path = environment.configFile().resolve("elasticsearch" + allowedSuffix);
+                    if (Files.exists(path)) {
+                        settingsBuilder.loadFromPath(path);
                     }
                 }
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: I'm wondering if we should fail if we find several times the config file with different extensions?

File: core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java [hunk 2]
```diff
@@ -114,10 +114,9 @@
             }
             if (loadFromEnv) {
                 for (String allowedSuffix : ALLOWED_SUFFIXES) {
-                    try {
-                        settingsBuilder.loadFromPath(environment.configFile().resolve("elasticsearch" + allowedSuffix));
-                    } catch (SettingsException e) {
-                        // ignore
+                    Path path = environment.configFile().resolve("elasticsearch" + allowedSuffix);
+                    if (Files.exists(path)) {
```
Comment (rmuir) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: I don't like Files.exists in general but personally I think this is the right decision for a good surgical fix!

File: core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: I think you're right, but I don't know the history of allowing multiple settings formats and whether or not it is used in practice. If we are going to make a breaking change on this, now (pre 2.0) is most definitely the time.

**Comment by rmuir** [comment]
+1, I think this is a good step.

File: core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java (hunk 1 above)
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: FWIW I've used this in the past for production ES clusters to have a set of common settings (elasticsearch.yml) and node-specific settings (elasticsearch.json) to merge two files with settings.

That said, I still think it's safer/better to remove this feature and fail if more than one config file is found. It reduces the complexity for reasoning where a setting came from.

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: >  still think it's safer/better to remove this feature and fail if more than one config file is found. It reduces the complexity for reasoning where a setting came from.

+1

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java]: I'll open a separate issue for it.

**Comment by jasontedor** [comment]
Thanks for helpful review and comments @jpountz, @dakrone and @rmuir. Pushed to master and back ported to 2.0. I've opened a separate issue #13042 to address multiple settings files.

**Pull request merged**